### PR TITLE
add a detection zone and trigger game end if you do not collid with a…

### LINKED ISSUE #3: add collision detection so that you must break breakable cubes

### PR DESCRIPTION
**Pull request by chriskarpyszyn**
… destructable object

Closes #3

### CODE REVIEW

File: Assets/Scripts/Destructable.cs [hunk 1]
```diff
@@ -10,7 +10,6 @@ void OnCollisionEnter(Collision collisionInfo)
     {
         if (collisionInfo.collider.tag == "Player")
         {
-            Debug.Log("Test");
             Instantiate(destroyedVersion, transform.position, transform.rotation);
             Destroy(gameObject);
         }
```
Comment (asim-shrestha) [inline Assets/Scripts/Destructable.cs:15]: A comment below this says `//todo-ck add a timer to 'destroy' the pieces when they're offscreen`

You probably already knew this but a possibly better alternative is to just make your `DetectionZone` 'garbage collect' these already destroyed destructables

File: Assets/Scripts/PlayerCollision.cs [hunk 2]
```diff
@@ -5,16 +5,15 @@ public class PlayerCollision : MonoBehaviour
...
 
-    //will be called on collision if object as a rigid body and a collider
+    //will be called on collision if object has a rigid body and a collider
     void OnCollisionEnter(Collision collisionInfo)
     {
 
         if (collisionInfo.collider.tag == "Obstacle")
         {
-            movement.enabled = false;
+            movement.DisableMovement();
```
Comment (asim-shrestha) [inline Assets/Scripts/PlayerCollision.cs:14]: Real minor but this method is also called within `FindObjectOfType<GameManager>().EndGame();`

File: Assets/Level 1.unity [hunk 3]
```diff
@@ -1617,6 +1617,131 @@ MonoBehaviour:
   m_Name: 
   m_EditorClassIdentifier: 
   movement: {fileID: 1630498347}
+--- !u!1 &1731547812
```
Comment (asim-shrestha) [inline Assets/Level 1.unity:1620]: The woes of git + unity are files like these 😭

**Review by asim-shrestha — APPROVED** [review]
LGTM!

File: Assets/Scripts/Destructable.cs (hunk 1 above)
Comment (chriskarpyszyn) [reply Assets/Scripts/Destructable.cs:15]: I didn't think of that actually but it makes total sense! Thanks!

File: Assets/Scripts/PlayerCollision.cs (hunk 2 above)
Comment (chriskarpyszyn) [reply Assets/Scripts/PlayerCollision.cs:14]: good catch

File: Assets/Level 1.unity (hunk 3 above)
Comment (chriskarpyszyn) [reply Assets/Level 1.unity:1620]: Yeah, I'm just moving things around in the hierarchy.

**Pull request merged**